Add Examples to show the oracledb module functionalities (#36)
ballerina-platform/module-ballerinax-oracledb: examples/online-shop/README.md

# Ballerina OracleDB module Example use case

## Overview 
This example demonstrates how to use the ballerina oracledb module to execute, query and call etc. in an oracle database. 

> 

Reviewer: Shall we write an overview of the use case(scenario) we implemented here

Author: Updated in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/36/commits/726dbeffdcc1c1f0c1df33b9bd7e62f715f4afe5

# Ballerina OracleDB module Example use case

> ## overview 

Reviewer 2: ```suggestion
## Overview 
```

Author: corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/36/commits/fe7671fcf8c033cd93419840a549cb2dd8ad96d1

This example demonstrates how to use the ballerina oracledb module to execute, query and call etc. in an oracle database. 

## Implementation

> These are HTTP RESTful services that are used to insert, create, and retrieve data of a sample database.

Reviewer 2: ```suggestion
These are HTTP RESTful services used to insert, create, and retrieve data of a sample database.
```

Author: corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/36/commits/fe7671fcf8c033cd93419840a549cb2dd8ad96d1

# Ballerina OracleDB module Example use case

## Overview 

> This example demonstrates how to use the ballerina oracledb module to execute, query and call etc. in an oracle database. 

Reviewer 3: ```suggestion
This example demonstrates how to use the Ballerina `oracledb` module to execute, query, and call etc. in an Oracle database. 
```

> Here a sample database is used to demonstrate the functionalities. This sample database models an imaginary shop that sells

Reviewer 3: ```suggestion
Here, a sample database is used to demonstrate the functionalities. This sample database models an imaginary shop that sells
```

> computer hardware and database have six tables namely `CUSTOMERS`, `CONTACTS`, `EMPLOYEES`, `PRODUCTS`, `ORDERS` and `ORDERITEMS`.

Reviewer 3: ```suggestion
computer hardware, and the database has six tables namely, `CUSTOMERS`, `CONTACTS`, `EMPLOYEES`, `PRODUCTS`, `ORDERS`, and `ORDERITEMS`.
```

> The shop have customer information including name, address etc. with their respective contact details. It also maintains an employee list

Reviewer 3: ```suggestion
The shop has customer information including the name, address, etc. with their respective contact details. It also maintains an employee list
```

> of the shop. Once a customer place an order with different order items chose from the products list, the shop can follow and update 

Reviewer 3: ```suggestion
of the shop. Once a customer places an order with different order items chosen from the products list, the shop can follow up and update 
```

> the order status to several stages like `Pending`, `Shipped` or `Cancelled`.

Reviewer 3: ```suggestion
the order status in several stages like `Pending`, `Shipped`, or `Cancelled`.
```

## Implementation

These are HTTP RESTful services used to insert, create, and retrieve data of a sample database.

## Prerequisite

* *Adding the oracledb JDBC thin driver and their dependencies*

>     * Download and add the OracleDB thin driver `ojdbc8.jar` along with `xdb.jar` and `xmlparserv2.jar` as native 

Reviewer 3: ```suggestion
    * Download and add the OracleDB thin driver `ojdbc8.jar` along with the `xdb.jar` and `xmlparserv2.jar` files as native 
```

>       library dependencies in the example Ballerina project's `Ballerina.toml` file. It is recommended to use an oracle 

Reviewer 3: ```suggestion
      library dependencies in the example Ballerina project's `Ballerina.toml` file. It is recommended to use an Oracle 
```

      thin driver `ojdbc8.jar` version greater than 12.2.0.1. As per the existing `Ballerina.toml` file, it points to 

>       those dependencies in `ballerina` folder. You may have to download those dependencies and update the `Ballerina.toml`

Reviewer 3: ```suggestion
      those dependencies in the `ballerina` folder. You may have to download those dependencies and update the `Ballerina.toml`
```

      file.
      
      Follow one of the ways below to add the JAR in the file: 
      
      * Download the JAR files and update the path.
        ```
        [[platform.java11.dependency]]
        path = "PATH"
        ```
        or
        

>       * Add JAR with the Maven dependency params.

Reviewer 3: ```suggestion
      * Add the JAR with the Maven dependency params.
```

        ```
        [platform.java11.dependency]]
        groupId = "com.oracle.database.jdbc"
        artifactId = "ojdbc8"
        version = "12.2.0.1"
  
        [platform.java11.dependency]]
        groupId = "com.oracle.database.xml"
        artifactId = "xdb"
        version = "21.1.0.0"
  
        [platform.java11.dependency]]
        groupId = "com.oracle.database.xml"
        artifactId = "xmlparserv2"
        version = "12.2.0.1"
        ```

> * *Initial schema, table creation and data insertion*

Reviewer 3: ```suggestion
* *Initial schema, table creation, and data insertion*
```

>     * Start and run the oracle database docker container or standalone database.

Reviewer 3: ```suggestion
    * Start and run the Oracle database Docker container or standalone database.
```

>     * Run `schema_with_data.sql` script in oracle database.

Reviewer 3: ```suggestion
    * Run the `schema_with_data.sql` script against the Oracle database.
```

>         * Goto `setup` folder of the `examples/onlineshop` directory and open the `schema_with_data.sql` file.

Reviewer 3: ```suggestion
        * Navigate to the `setup` folder of the `examples/onlineshop` directory, and open the `schema_with_data.sql` file.
```

>         * Run the content of the file in oracle database. 

Reviewer 3: ```suggestion
        * Run the content of the file in the Oracle database. 
```

    
## Run the Example

> To start the service, move into the `examples/onlineshop` folder and execute the command below.

Reviewer 3: ```suggestion
To start the service, navigate to the `examples/onlineshop` folder, and execute the command below.
```
